platform/o.n.bootstrap/src/org/netbeans/NbClipboard.java

```diff
@@ -419,12 +422,34 @@ private final class GetContents implements Runnable {
         public GetContents(boolean notify) {
             this.notify = notify;
         }
-        
+
         @Override
+        @SuppressWarnings("SleepWhileInLoop")
         public void run() {
-            log.fine("Running update");
+            log.fine("Running update"); // NOI18N
             try {
-                Transferable transferable = systemClipboard.getContents(this);
+                Transferable transferable = null;
+                // There can be a race between multiple applications accessing
+                // the clipboard. If access can't be optained directly, retry
+                // for a maximum of 1s. This is called from the requestprocessor
```

Comment:
like I mentioned on the other PR: maximum of 1s implies that `getContents` returns immediately on exception. Is this the case?

If this isn't certain that all implementations return immediately, the loop could be modified to include a `(System.currentTimeMillis()-start) < 1_000)` condition.

like common for networking:
https://github.com/apache/netbeans/blob/f48f91e6c197d8a40bd82fc2f2d12a4e71242afe/java/maven.indexer/src/org/netbeans/modules/maven/indexer/SMORequestResult.java#L90-L92

Reply:
Oh, good point. I've seen old BugZilla tickets indicating that Clipboard calls can take a while to return in some cases.

Reply:
@eirikbakke that's the reason it's off the EDT in the first place.  I think it's likely that it will return immediately when not available, but possibly better not to rely on it.  At the same time, EDT code already times out on the wait.

The exception (on Windows) comes from here btw - https://github.com/openjdk/jdk/blob/master/src/java.desktop/windows/native/libawt/windows/awt_Clipboard.cpp#L138

Reply:
thanks for digging into the jdk code, the win API doc of that function also doesn't mention anything:
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-openclipboard#remarks

so it is probably ok to assume that the call returns quickly on failure. I am just sometimes overly careful with things like that, since you never know how the calls are synchronized etc. (esp when I don't know anything about the impl)